error_113.go

```diff
@@ -2,6 +2,38 @@
 
 package errorx
 
+import "reflect"
+
+// todo add errorx.As()?
+// todo make another 'go error type' for passing into errors.As()?
+// todo when fixed, add tests for wrap/decorate etc.
+// As checks if target is of the same type as current error and, if true, sets target to this error value.
+// NB: Call to errors.As() converts any type of errorx error to any other type,
+// therefore such calls are currently unsafe for errorx errors and will likely break semantics.
+// Note than calls to errors.Is() do not suffer from the same issue.
+func (e *Error) As(target interface{}) bool {
+	targetWrapper, wrapperOk := target.(*typeCheckTarget)
+	if wrapperOk && !e.Is(targetWrapper.err) {
+		return false
+	}
+	if !wrapperOk  {
+		targetError, ok := target.(*error)
```

Comment:
`ok` should always be `false` because any error is assignable to `error` type. See https://github.com/golang/go/blob/go1.16/src/errors/wrap.go#L91-L94